### DISCUSSION TITLE
Could a 8086/V30 version of this be developed for the OlivettI M24 (AT&T 6300), the IBM PS/2 Model 30, etc?

### DISCUSSION THREAD

**Discussion by volkertb** [640-KB/GLaBIOS#13] [Suggestions]
I love seeing new software (and firmware) being developed to squeeze every last bit of performance out of our beloved vintage systems, and to allow them to do things they were never designed to do. Thanks for sharing this with the world! :relaxed:

How difficult would it be to port `GLaBIOS` to popular 8086/V30 systems from back in the day?

There are a number notable of 8086-based PC/XT systems that this would be useful for:

- Olivetti M24 / AT&T 6300
- IBM PS/2 Models 25 and 30 (there are both 8086-based and 286-based variants of these models)
- 8086-based models of the Tandy 1000

For instance, I grew up with the 8086-based Olivetti M24, and it's always been an obsession for me to tune and optimize it further than the designers of the system initially intended. Would love to see a version of GLaBIOS being developed for that old beast.

Thanks for considering.

**Comment by 640-KB** [comment]
Hi @volkertb! I've moved this to the Ideas discussion so we can collect ideas. This has definitely been something that's been discussed and something that would be great to see.  In fact, we've been discussing what it might require to run on an Amstrad PC1512 or PC6400 - very similar to the M24/6300.

Generally speaking, the 8086 CPU doesn't represent a significant difference to the BIOS itself, however many of those upgraded PCs that you mentioned (Tandy, PS/2, etc) also added additional hardware (custom video adapters/displays, realtime clocks, CMOS, etc) that would require (possibly significant) new code to be written to support.  The original intent of this BIOS is for it to be a small drop-in alternative for clone PCs that use the smallest 8K of ROM code.

That said, adding basic support for those chips, so long as it doesn't adversely affect other code or require significant code changes is completely within reason.  The plan is to keep the "core" BIOS general and over time develop and release "Companion ROMs" to provide supplemental support for other vintage systems.  Looking for contributors to help with that so if you know anyone who can help, please spread the word!

Thanks for the suggestions and we do plan to get there!

640KB
